tests/test_pwhash.py

```diff
@@ -86,6 +109,20 @@ def gen_argon2i_str_params(n,
     return rn
 
 
+def gen_argon2id_str_params(n,
```

Comment:
We should consider using hypothesis for this type of property based testing.

Reply:
Do you think we could defer the change to some later PR (my preferred option), or should we add the build-time dependency and adapt the tests here?

Reply:
As suggested, I've replaced the homegrown generator with use of hypothesis property generation.
I've set a very long ``deadline`` time, and a small number of test cases to keep test timing within reasonable limits.